extensions/positron-run-app/src/api.ts

```diff
@@ -168,44 +186,45 @@ export class PositronRunAppApiImpl implements PositronRunApp, vscode.Disposable
 			() => log.warn('Timed out waiting for existing terminals to close. Proceeding anyway'),
 		);
 
-		// Create a disposables store for this session.
+		const shellIntegration = await raceTimeout(
+			shellIntegrationPromise,
+			shellIntegrationTimeout,
+			() => log.warn('Timed out waiting for shell integration to be ready'),
+		);
 
-		// Create a promise that resolves when the server URL has been previewed,
-		// or an error has occurred, or it times out.
-		const didPreviewUrl = raceTimeout(
-			new Promise<boolean>((resolve) => {
-				const disposable = vscode.window.onDidStartTerminalShellExecution(async e => {
-					// Remember that shell integration is supported.
-					await this.setShellIntegrationSupported(true);
+		progress.report({ message: vscode.l10n.t('Starting application...') });
 
-					if (e.terminal === terminal) {
-						const previewOptions: AppPreviewOptions = {
-							proxyInfo,
-							urlPath: options.urlPath,
-							appReadyMessage: options.appReadyMessage,
-							appUrlStrings: options.appUrlStrings,
-						};
-						const didPreviewUrl = await previewUrlInExecutionOutput(e.execution, previewOptions);
-						if (didPreviewUrl) {
-							resolve(didPreviewUrl);
-						}
-					}
-				});
-				this._runApplicationDisposableByName.set(options.name, disposable);
-			}),
-			didPreviewUrlTimeout,
-			async () => {
-				await this.setShellIntegrationSupported(false);
-			});
+		if (shellIntegration) {
+			log.info('Shell integration is supported. Executing command with shell integration.');
 
-		// Execute the command.
-		progress.report({ message: vscode.l10n.t('Starting application...') });
-		terminal.sendText(terminalOptions.commandLine, true);
+			// Remember that shell integration is supported.
+			await this.setShellIntegrationSupported(true);
+
+			// Execute the command.
+			const execution = shellIntegration.executeCommand(terminalOptions.commandLine);
+
+			// Wait for the server URL in the execution output.
+			const previewOptions: AppPreviewOptions = {
+				proxyInfo,
+				urlPath: options.urlPath,
+				appReadyMessage: options.appReadyMessage,
+				appUrlStrings: options.appUrlStrings,
+			};
+			await previewUrlInExecutionOutput(execution, previewOptions);
+		} else {
+			log.info('Shell integration not supported. Executing command without shell integration.');
 
-		if (isShellIntegrationEnabledAndSupported && !await didPreviewUrl) {
-			log.warn('Failed to preview URL using shell integration');
 			// TODO: If a port was provided, we could poll the server until it responds,
 			//       then open the URL in the viewer pane.
+
+			// Execute the command without shell integration.
+			terminal.sendText(terminalOptions.commandLine, true);
+
+			// Remember that shell integration is not supported to display the guide in future runs.
+			await this.setShellIntegrationSupported(false);
+
+			// Guide the user to use a supported shell.
+			showShellIntegrationNotSupportedMessage().catch(() => { });
```

Comment:
```suggestion
			await showShellIntegrationNotSupportedMessage().catch(() => { });
```

Reply:
The promise would only resolve when the user clicks a button in the notification, and I don't think we want to tie the `runApplication` promise to that